main.go

```diff
@@ -84,23 +95,51 @@ func main() {
 		glog.Fatal(err)
 	}
 
-	var metricConfig *util.MetricConfig
-	if *enableMetrics {
-		metricConfig = &util.MetricConfig{
-			MetricsEndpoint: *metricsEndpoint,
-			MetricsPort:     *metricsPort,
-			MetricsPrefix:   *metricsPrefix,
-			MetricsLabels:   metricsLabels,
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+
+	stopCh := make(chan struct{}, 1)
+	startCh := make(chan struct{}, 1)
+
+	if *enableLeaderElection {
+		hostname, err := os.Hostname()
+		if err != nil {
+			glog.Fatal(err)
+		}
+		resourceLock, err := resourcelock.New(resourcelock.ConfigMapsResourceLock, *leaderElectionLockNamespace, *leaderElectionLockName, kubeClient.CoreV1(), resourcelock.ResourceLockConfig{
+			Identity: hostname,
+			// TODO: This is a workaround for a nil dereference in client-go. This line can be removed when that dependency is updated.
```

Comment:
Any pointer to some github issue that explains the issue?

Reply:
It was fixed a while ago, the client-go dependency is just a couple versions behind: https://github.com/kubernetes/client-go/blame/b831b8de7155117e51afaffeb647007a756ddc92/tools/leaderelection/resourcelock/configmaplock.go#L96-L98

Reply:
Which version of `client-go` has the fix?

Reply:
`kubernetes-1.14.0-alpha.1` and later, looks like

Reply:
In that case, should we actually supply a real `EventRecorder`, like the one used by the `SparkApplication` controller?

Reply:
I think we can also update the version of `client-go` and other k8s libs.